src/porepy/numerics/ad/operators.py

```diff
@@ -1234,15 +1132,228 @@ def _parse_other(self, other):
+    representation of an operator instance at a previous time.
+
+    Operators created via constructor always start at the current time.
+
+    """
+
+    def __init__(
+        self,
+        name: str | None = None,
+        domains: Optional[pp.GridLikeSequence] = None,
+        operation: Optional[Operator.Operations] = None,
+        children: Optional[Sequence[Operator]] = None,
+    ) -> None:
+        super().__init__(
+            name=name, domains=domains, operation=operation, children=children
+        )
+
+        self.original_operator: Operator
+        """Reference to the operator representing this operator at the current time amd
+        iterate.
+
+        This attribute is only available in operators representing previous time steps.
+
+        """
+
+        self._time_step_index: int = -1
+        """Time step index, starting with -1 (current time) and increasing for previous
+        time steps."""
+
+    @property
+    def prev_time(self) -> bool:
+        """True, if the operator represents a previous time-step."""
+        return True if self._time_step_index > -1 else False
+
+    @property
+    def time_step_index(self) -> Optional[int]:
+        """Returns the time step index this instance represents.
+
+        - None indicates this is an operator at the current time step
+        - 0 represents the first previous time step
+        - 1 represents the next time step further back in time
+        - ...
+
+        Note:
+            The public time step index is ``None``, if the operator represents the
+            current time.
+
+        """
+        return self._time_step_index if self._time_step_index >= 0 else None
+
+    def previous_timestep(
+        self: _TimeDependentOperator, steps: int = 1
+    ) -> _TimeDependentOperator:
+        """Returns a copy of the time-dependent operator with an advanced time-step
+        index.
+
+        Time-dependent operators do not invoke the recursion (like the base class),
+        but represent a leaf in the recursion tree.
+
+        Note:
+            You cannot create operators at the previous time step from operators which
+            are at some previous iterate. Use the :attr:`original_operator` instead.
+
+        Parameters:
+            steps: ``default=1``
+
+                Number of steps backwards in time.
+
+        Raises:
+            ValueError: If this instance represents an operator at a previous iterate.
+            AssertionError: If ``steps`` is not strictly positive.
+
+        """
+        if isinstance(self, IterativeOperator):
+            if self.prev_iter:
+                raise ValueError(
+                    "Cannot create an operator representing a previous time step,"
+                    + " if it already represents a previous iterate."
+                )
+
+        assert steps > 0, "Number of steps backwards must be strictly positive."
+        # TODO copy or deepcopy? Is this enough for every operator class?
```

Comment:
Since this would require changes in several files (storing discr. matrices with time or iterate indices), I would suggest to put this in a `Note` in the module documentation. Also to notify users.

Reply:
Let's not overcomplicate things, but leave it as it is for now.